src/otx/cli/utils/experiment.py

```diff
@@ -0,0 +1,253 @@
+"""Utils function for experiments."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import logging
+import multiprocessing as mp
+import os
+import time
+from abc import ABC, abstractmethod
+from pathlib import Path
+from statistics import mean
+from typing import Any, Dict, List, Optional, Union, no_type_check
+
+import psutil
+import yaml
+
+try:
+    import pynvml
+except ImportError:
+    pynvml = None
+
+logger = logging.getLogger(__name__)
+GIB = 1024**3
+AVAILABLE_RESOURCE_TYPE = ["cpu", "gpu"]
+
+
+class ResourceTracker:
+    """Class to track resources usage.
+
+    Args:
+        resource_type (str, optional): Which resource to track. Available values are cpu, gpu or all now.
+                                        Defaults to "all".
+        gpu_ids (Optional[str]): GPU indices to record.
+    """
+
+    def __init__(self, resource_type: str = "all", gpu_ids: Optional[str] = None):
+        if resource_type == "all":
+            self._resource_type = AVAILABLE_RESOURCE_TYPE
+        else:
+            self._resource_type = [val for val in resource_type.split(",")]
+
+        gpu_ids_arr = None
+        if gpu_ids is not None:
+            gpu_ids_arr = [int(idx) for idx in gpu_ids.split(",")]
+            gpu_ids_arr[0] = 0
```

Comment:
Why do we need to specify the first element of gpu id array?

Reply:
In multi GPU training case, main process always uses `0` index GPU. it's applied to that code.

Reply:
What if user gives "1,2" as a gpu_ids? Should gpu_ids_arr be [0,2]?

Reply:
Yes you're right. I tried to change it before, it's quite hard, because origin solution is setting `CUDA_VISIBLE_DEVICES` before executing python file. I'll try to find other ways to make it run as we expect. After than I can change this code also.

Reply:
If this is intended behavior, then it is okay to leave it as it is.